lens-server/src/main/java/org/apache/lens/server/query/QueryExecutionServiceImpl.java

```diff
@@ -2054,8 +2056,14 @@ public QueryPrepareHandle prepare(LensSessionHandle sessionHandle, String query,
       acquire(sessionHandle);
       prepared = prepareQuery(sessionHandle, query, lensConf, SubmitOp.PREPARE);
       prepared.setQueryName(queryName);
-      prepared.getSelectedDriver().prepare(prepared);
+      try {
```

Comment:
For the prepared query, I am inserting start_time and timeTaken(this can deduce the end_time if needed).
Regarding the NumberOfPrepared queries, it is already being emitted PREPARED_QUERIES_COUNTER.  Regarding the query prepare time is already present in  database.

Reply:
This is already happening. All the public methods of QueryServiceResource histograms are already pushed by default.